test/boost/mutation_test.cc

```diff
@@ -2531,6 +2531,37 @@ SEASTAR_THREAD_TEST_CASE(test_mutation_consume) {
     }
 }
 
+class validating_consumer {
+    mutation_fragment_stream_validator _validator;
+
+public:
+    explicit validating_consumer(const schema& s) : _validator(s) { }
+
+    void consume_new_partition(const dht::decorated_key&) {
+        BOOST_REQUIRE(_validator(mutation_fragment_v2::kind::partition_start, position_in_partition_view(position_in_partition_view::partition_start_tag_t{})));
+    }
+    void consume(tombstone) { }
+    stop_iteration consume(static_row&& sr) {
+        BOOST_REQUIRE(_validator(mutation_fragment_v2::kind::static_row, sr.position()));
+        return stop_iteration::no;
+    }
+    stop_iteration consume(clustering_row&& cr) {
+        BOOST_REQUIRE(_validator(mutation_fragment_v2::kind::clustering_row, cr.position()));
+        return stop_iteration::no;
+    }
+    stop_iteration consume(range_tombstone_change&& rtc) {
+        BOOST_REQUIRE(_validator(mutation_fragment_v2::kind::range_tombstone_change, rtc.position()));
+        return stop_iteration::no;
+    }
+    stop_iteration consume_end_of_partition() {
+        BOOST_REQUIRE(_validator(mutation_fragment_v2::kind::partition_end, position_in_partition_view(position_in_partition_view::end_of_partition_tag_t{})));
+        return stop_iteration::no;
+    }
+    void consume_end_of_stream() {
+        BOOST_REQUIRE(_validator.on_end_of_stream());
+    }
+};
+
```

Comment:
it seems appropriate to move this to test/lib/ maybe to a new header file e.g. mutation_fragment_stream_validator.hh
so it can be used also elsewhere down the road.

Reply:
or to mutation_assertions.hh that already calls BOOST_* stuff.